src/lambda_streams/Animals.java

```diff
@@ -26,36 +28,110 @@ public static void main(String[] args) {
         List<String> lowered = lowerFirst(animals,false);
         System.out.println(lowered);
 
-        
+        System.out.println("***************");
+
+        List<String> flippedClone = flipAnimals(false);
+        System.out.println(flippedClone);
+
+        List<String> flipped = flipAnimals(true);
+        System.out.println(flipped);
+
+        System.out.println("***************");
+
+        List<String> sortedClone = sortAnimals(false);
+        System.out.println(sortedClone);
+
+        List<String> sorted = sortAnimals(true);
+        System.out.println(sorted);
 
     }
 
     static List<String> capsFirst(List<String> animaList, boolean mutate) {
-        //clean up the animals list so that the first letter is capitalized, and all the other letters are lowercased. Use a stream to accomplish this task.  Also, the 2nd parameter of this function is a boolean.  use this boolean 'flag' to determine whether or not to 'mutate' the original animals array stored as a static class field.  if the flag is set to 'true', mutate the animals and return the animals out of the function.  if it is false, create a copy of the animals, perform your stream operations on the copy, and return the copy of animals out of the function, WITHOUT modifying the original animals array.  
+        //clean up the animals list so that the first letter is capitalized, and all the other letters are lowercased.
+        // Use a stream to accomplish this task.  Also, the 2nd parameter of this function is a boolean.  use this
+        // boolean 'flag' to determine whether or not to 'mutate' the original animals array stored as a static class
+        // field.  if the flag is set to 'true', mutate the animals and return the animals out of the function.  if it
+        // is false, create a copy of the animals, perform your stream operations on the copy, and return the copy of
+        // animals out of the function, WITHOUT modifying the original animals array.
+        if(mutate){
+            animals  = animaList.stream().map(c -> c.substring(0,1).toUpperCase()+ c.substring(1).toLowerCase())
+                    .collect(Collectors.toList());
+        }else{
+            List<String> newList = animaList.stream().map(c -> c.substring(0,1).toUpperCase()+ c.substring(1).toLowerCase())
+                    .collect(Collectors.toList());
+            animals = new ArrayList<String>(newList);
```

Comment:
This line replaces the 'animals' list with the new capitalized list, thought the 'mutate' value is "false".